apollo-annotations/src/appleMain/kotlin/com/apollographql/apollo3/annotations/ApolloAdaptableWith.kt

```diff
@@ -0,0 +1,10 @@
+package com.apollographql.apollo3.annotations
+
+import kotlin.reflect.AssociatedObjectKey
+import kotlin.reflect.ExperimentalAssociatedObjects
+import kotlin.reflect.KClass
+
+@OptIn(ExperimentalAssociatedObjects::class)
```

Comment:
This is using experimental Kotloin stdlib features but since it's only for native/JS, I think it's ok